src/product/product.service.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { validate as isUUID } from 'uuid';
+
+import { CreateProductDto } from './dto';
+import { Product } from './entities/product.entity';
+import { PaginationDto } from '../common/dto/pagination.dto';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<Product>,
```

Comment:
WDYT about creating a product repository file to handle all db queries? I´ll explain it on slack

Reply:
I've added a custom repository as discussed